torchrec_dlrm/dlrm_main.py

```diff
@@ -499,75 +406,54 @@ class TrainValTestResults:
 
 def train_val_test(
     args: argparse.Namespace,
-    train_pipeline: TrainPipelineSparseDist,
+    pipeline: TrainPipelineSparseDist,
```

Comment:
since we will be creating new pipelines inside here, IMO it makes more sense to create the main train_pipeline in this function as well (instead of passing in train_pipeline and creating more pipelines inside)